docs/src/complex.md

```diff
@@ -56,3 +56,40 @@ julia> wirtinger(x -> 3x^2 + 2x + 1, 1+2im)
 julia> wirtinger(x -> abs2(x), 1+2im)
 (1.0 - 2.0im, 1.0 + 2.0im)
 ```
+
+The gradient definition Zygote uses can also be expressed in terms of the [Wirtinger calculus](https://en.wikipedia.org/wiki/Wirtinger_derivatives) using the operators ``\frac{\partial}{\partial z}`` and ``\frac{\partial}{\partial z^*}``:
+
+```math
+f: \mathbb{C} \rightarrow \mathbb{R}, \qquad
+f(z) \equiv f(x + iy), \qquad
+z \in \mathbb{C}, \ x, y \in \mathbb{R} \\[1.2em]
+
+\bar f \equiv \frac{\partial f}{\partial x} + i \frac{\partial f}{\partial y}
+    = 2 \, \frac{\partial(\mathrm{Re}(f))}{\partial z^*}
+    = \frac{\partial(f + f^*)}{\partial z^*}
+    = \frac{\partial f}{\partial z^*} + \left(\frac{\partial f}{\partial z}\right)^{\!*}
+```
+
+For the composition of two function, one gets the following pullback map, if the inner function is holomorphic:
+
+```math
+f: \mathbb{C} \rightarrow \mathbb{R}, \qquad
+w: \mathbb{C} \rightarrow \mathbb{C} \\[1.2em]
+
+\begin{align*}
+\overline{f \circ w}
+    &= \frac{\partial (f \circ w)}{\partial z^*} + \left(\frac{\partial (f \circ w)}{\partial z}\right)^{\!*}
+    = \frac{\partial f}{\partial w} \frac{\partial w}{\partial z^*}
+        + \frac{\partial f}{\partial w^*} \frac{\partial w^*}{\partial z^*}
+        + \left( \frac{\partial f}{\partial w} \frac{\partial w}{\partial z}
+        + \frac{\partial f}{\partial w^*} \frac{\partial w^*}{\partial z} \right)^{\!*} \\
+    &= \left[ \frac{\partial f}{\partial w^*} + \left( \frac{\partial f}{\partial w} \right)^{\!*} \right]
+        \left( \frac{\partial w}{\partial z} \right)^{\!*}
+    = \overline{f} \cdot \left( \frac{\partial w}{\partial z} \right)^{\!*}
+    \qquad \text{if $w(z)$ holomorphic} \Leftrightarrow \frac{\partial w}{\partial z^*} = 0
+\end{align*}
```

Comment:
I basically see what the above section is saying (though it might be nice to have an implementation in code for the sake of precision/clarity). This section strikes me as confusing though; a big equation dump there and while I'm sure it's all correct, I'm not sure what it's trying to get across.

Reply:
I can try explaining this a bit better in words. I'm trying to give the mathematical reason here, why we need to take the complex conjugate of the derivative in pullbacks.